xarray/conventions.py

```diff
@@ -490,8 +490,9 @@ def decode_cf_variable(name, var, concat_characters=True, mask_and_scale=True,
         del attributes['dtype']
         data = BoolTypeArray(data)
 
-    return Variable(dimensions, indexing.LazilyOuterIndexedArray(data),
-                    attributes, encoding=encoding)
+    if not isinstance(data, dask_array_type):
+        data = indexing.LazilyOuterIndexedArray(data)
```

Comment:
This is the offending line which caused the dask arrays to stop being dask arrays. I have simply bypassed it for dask arrays. This didn't break any tests. But can we think of any unforeseen negative consequences? What does `LazilyOuterIndexedArray` do here? I have to admit that I don't understand the details of our new indexing wrappers.

Reply:
`LazilyOuterIndexedArray` makes indexing on non-dask arrays lazy. It's not necessary or useful for dask arrays.